scripts/postgres18-verify-and-finalize.sh

```
@@ -0,0 +1,365 @@
+      wait_for_ready "$OLD_CONTAINER_NAME" "$POSTGRES_USER" "postgres"
+      started_from_old_data=1
+    else
+      log "Old data directory could not be started cleanly; falling back to logical backup verification."
+      docker logs "$OLD_CONTAINER_NAME" >&2 || true
+      docker rm -f "$OLD_CONTAINER_NAME" >/dev/null 2>&1 || true
+    fi
+  fi
+fi
+
+if (( started_from_old_data == 0 )); then
+  latest_backup="$(ls -1t "$BACKUP_DIR"/pg_dumpall_*.sql 2>/dev/null | head -1 || true)"
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)